contracts/PetPark.sol

```diff
@@ -1,7 +1,87 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
+    address public owner;
+
+    enum Pets {
+        None,
+        Fish,
+        Cat,
+        Dog,
+        Rabbit,
+        Parrot
+    }
+    enum Gender {
+        Male,
+        Female
+    }
+    struct Person {
+        Gender gender;
+        uint256 age;
+        Pets animal;
+        bool isExists;
+        bool isBorrowed;
+    }
+
+    mapping(Pets => uint256) public animalCounts;
+    mapping(address => Person) public person;
+
+    event Added(Pets _pet, uint256 _count);
+    event Borrowed(Pets _pet);
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function add(Pets _animal, uint256 _count) public {
+        require(msg.sender == owner, "Not an owner");
+        require(_animal != Pets.None, "Invalid animal");
+        animalCounts[_animal] += _count;
+        emit Added(_animal, _count);
+    }
+
+    function borrow(
+        uint256 _age,
+        Gender _gender,
+        Pets _animal
+    ) public {
+        require(_age > 0, "Invalid Age");
+        require(_animal != Pets.None, "Invalid animal type");
+        require(animalCounts[_animal] != 0, "Selected animal not available");
+
+        if (person[msg.sender].isExists) {
+            require(person[msg.sender].age == _age, "Invalid Age");
```

Comment:
You can store `person[msg.sender]` in a separate memory variable and use that whenever you need to make reads. Later in the course you will learn that it is a gas optimisation technique